src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/RequestHandler.java

```diff
@@ -2,44 +2,98 @@
         Boolean passported = null;
-        Outcome oc = null;
+        Outcome outcome = null;
 
-        if (request.getSectionUnder18() != null) {
-            under18 = request.getSectionUnder18().getClientUnder18();
+        if (cfeCrimeRequest.getSectionUnder18() != null) {
+            under18 = cfeCrimeRequest.getSectionUnder18().getClientUnder18();
         }
-        if (request.getSectionPassportedBenefit() != null) {
-            passported = request.getSectionPassportedBenefit().getPassportedBenefit();
+        if (cfeCrimeRequest.getSectionPassportedBenefit() != null) {
+            passported = cfeCrimeRequest.getSectionPassportedBenefit().getPassportedBenefit();
         }
-        oc = getOutcomeFromAgeAndPassportedBenefit(under18,passported);
+        outcome = getOutcomeFromAgeAndPassportedBenefit(under18, passported);
+
+        CfeCrimeResponse cfeCrimeResponse = new CfeCrimeResponse();
+        if (outcome != null) {
+            cfeCrimeResponse.setOutcome(outcome);
+        }else{
+            //call cma
+            statelessApiResponse = cmaService.callCma(cmaRequest);
+
+            setInitialMeansTestOutcome(cfeCrimeRequest, cfeCrimeResponse);
+            if (statelessApiResponse.getInitialMeansAssessment().getResult() == InitAssessmentResult.FULL) {
+                setFullMeansTestOutcome(cfeCrimeRequest, cfeCrimeResponse);
+            }
 
-        CfeCrimeResponse response = new CfeCrimeResponse();
-        if (oc != null) {
-            Result result = new Result();
-            result.setOutcome(oc);
-            response.withResult(result);
         }
-        return response;
+        return cfeCrimeResponse;
     }
 
-    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit){
+    private static Outcome getOutcomeFromAgeAndPassportedBenefit(Boolean clientUnder18, Boolean clientPassportedBenefit) {
         Outcome outcome = null;
-        if (clientUnder18 != null && clientUnder18.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientUnder18 != null && clientUnder18.booleanValue()) {
+            outcome = Outcome.ELIGIBLE;
         }
-        if (clientPassportedBenefit !=null && clientPassportedBenefit.booleanValue()){
-            outcome = Result.Outcome.ELIGIBLE;
+        if (clientPassportedBenefit != null && clientPassportedBenefit.booleanValue()) {
+            outcome = Outcome.ELIGIBLE;
         }
         return outcome;
     }
+
+    public static void setFullMeansTestOutcome(CfeCrimeRequest cfeCrimeRequest, CfeCrimeResponse cfeCrimeResponse) {
+
+        Outcome fullOutcome = null;
+
+        if (cfeCrimeRequest.getSectionFullMeansTest() != null){
+            FullAssessmentResult statelessInitialResult = statelessApiResponse.getFullMeansAssessment().getResult();
+            fullOutcome = FullMeansTestOutcomeCalculator.getFullMeansTestOutcome(statelessInitialResult, caseType, magCourtOutcome);
```

Comment:
I don't understand how this code even compiles. statelessApiResponse is instance data, but we're in static context. We've been bitten (badly) by side effects in cfe-civil, so we're really keen to avoid them in cfe-crime